### PR TITLE
Fix #119: use Zulu Java on macOS arm64 runners.

### LINKED ISSUE #119: Mac CI/CD fails because of Java 8
https://github.com/actions/setup-java/issues/625 -> https://github.com/actions/setup-java/issues/625#issuecomment-2096064399

Probably caused by: https://github.blog/changelog/2024-04-01-macos-14-sonoma-is-generally-available-and-the-latest-macos-runner-image/

Probable solution:
```patch
-macos-latest
+macos-13 # Java 8 is not supported on macos-14+ (M1).
```

I saw now new issue related to this at https://github.com/actions/runner-images/issues

### PR DESCRIPTION
**Pull request by TWiStErRob**
Left Ubuntu intact, as it's pre-installed!
* Fixes #119

### CODE REVIEW

File: .github/workflows/merge-check.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ jobs:
       - name: "Set up Java"
         uses: actions/setup-java@99b8673ff64fbf99d8d325f52d9a5bdedb8483e9 # v4
         with:
-          distribution: 'temurin'
+          distribution: ${{ (runner.os == 'macOS' && runner.arch == 'ARM64') && 'zulu' || 'temurin' }}
```
Comment (BraisGabin) [inline .github/workflows/merge-check.yml:32]: Easier and we don't really care that much about this.
```suggestion
          distribution: 'zulu'
```

**Review by BraisGabin — APPROVED** [review]

File: .github/workflows/merge-check.yml (hunk 1 above)
Comment (TWiStErRob) [reply .github/workflows/merge-check.yml:32]: > Left Ubuntu intact, as it's pre-installed!

Using temurin is a no-op (aliasing env vars), while Zulu is always a network request and unzip.

Comment (TWiStErRob) [reply .github/workflows/merge-check.yml:32]: Alternative if you don't like the ternary, we can expand the matrix (quite simple).

Comment (BraisGabin) [reply .github/workflows/merge-check.yml:32]: Then we can merge this :)

**Pull request merged**